crates/bevy_asset/src/path.rs

```diff
@@ -180,10 +180,25 @@ impl<'a> From<&'a Path> for AssetPath<'a> {
 }
 
 impl<'a> From<PathBuf> for AssetPath<'a> {
-    fn from(path: PathBuf) -> Self {
-        AssetPath {
-            path: Cow::Owned(path),
-            label: None,
+    fn from(mut path: PathBuf) -> Self {
+        match path.file_name().map(ToOwned::to_owned) {
+            Some(os_str) => {
+                let mut parts = os_str
+                    .to_str()
+                    .expect("File name format is not valid unicode")
+                    .splitn(2, '#');
+                path.pop();
+                path = path.join(parts.next().expect("Path must be set."));
```

Comment:
Same with this.

Reply:
eh since `impl From<String> for AssetPath` has possibility to panic, I thought it is allowed.

Reply:
Really? That shouldn't be the case. Imo fallible `From` impl should be `TryFrom` instead.

Reply:
I tested some cases, `parts.next().expect()` part is infalliable.
That is also misunderstanding, I would add handle of `to_str().expect()`

Reply:
If this is infallible, I would change both cases to a `.expect()` and leave a comment above each usage explaining why it can't fail.

Reply:
Irrc, I tested `""` empty string case, and It returned two empty string and no panic